lib/puppet/ffi/posix/functions.rb

```diff
@@ -0,0 +1,26 @@
+require 'puppet/ffi/posix'
+
+module Puppet::FFI::Posix
+  module Functions
+
+    extend FFI::Library
+
+    ffi_convention :stdcall
+
+    # https://man7.org/linux/man-pages/man3/getgrouplist.3.html
+    # https://www.gnu.org/software/libc/manual/html_node/Reading-Persona.html
```

Comment:
This documentation is about `getgroups()`, I don't think we need it here

Reply:
Added it regarding the `gid_t` actually being a `uint` but I agree.